Feature/generic button component

ShayMashiah opened the pull request.

ShayMashiah committed:
feat: tailwind configuration and initialization

ShayMashiah committed:
feat: shudcn configuration and initialization

ShayMashiah committed:
fix: change tailwind version from 4.1.7 to 3.4.3

ShayMashiah committed:
feat: creation of generic button component and styling acorrding to type and size

Taloreff-hls commented on src/components/common/GenericButton.tsx:
```diff
@@ -0,0 +1,22 @@
+      {children}
+    </Button>
+  );
+};
```
i dont think we need this, you can override the button variants coming from shadcn in src/ui/button.tsx

Taloreff-hls commented on src/components/ui/button.tsx:
```diff
@@ -0,0 +1,62 @@
+        md: "w-[82px] h-[36px] pt-[10px] pb-[10px] pr-[16px] pl-[16px] gap-[4px] rounded-[4px]",
+        lg: "w-[82px] h-[40px] pt-[12px] pb-[12px] pr-[16px] pl-[16px] gap-[4px] rounded-[4px]",
+        icon: "size-9",
+      },
```
name and style the variants according to the figma (we have primary and secondary)
use the colors coming from the tailwind.config

ShayMashiah committed:
fix: change files to ts and add style guides

ShayMashiah committed:
fix: modify tailwind configuration file

ShayMashiah committed:
This branch based on tailwind config
Merge branch 'feature/tailwind_configuration' into feature/generic_button_component

ShayMashiah committed:
fix: remove generic button component, convert styling from tailwind config and style secondary button
